components/raftstore/src/store/fsm/peer.rs

```diff
@@ -994,29 +994,84 @@ where
     // the flashback-only command in this way, But for RW local reads which need
     // to be considered, we let the leader lease to None to ensure that local reads
     // are not executed.
-    fn on_prepare_flashback(&mut self, ch: Sender<bool>) {
+    fn on_prepare_flashback(&mut self, cb: Callback<EK::Snapshot>) {
+        let region_id = self.region().get_id();
         info!(
             "prepare flashback";
-            "region_id" => self.region().get_id(),
+            "region_id" => region_id,
+            "peer_id" => self.fsm.peer.peer_id(),
+        );
+        // check persist state
+        let state_key = keys::region_state_key(region_id);
+        if let Some(target_state) = self
+            .ctx
+            .engines
+            .kv
+            .get_msg_cf::<RegionLocalState>(CF_RAFT, &state_key)
+            .unwrap()
+        {
+            info!(
+                "Prepare Flashback, already has a flashback local state for {:?} ",
+                self.region(),
+            );
+            if target_state.get_flashback_state() == FlashbackState::Processing {
+                info!("region is already in flashback state");
+                // check memory state
+                if self.fsm.peer.flashback_state.is_none() {
```

Comment:
```suggestion
                if self.fsm.peer.flashback_state.is_some() {
```

Reply:
This means that if the memory state does not exist, a callback will be called to move to the next stage: wait_apply.